Sources/Datadog/Core/Upload/DataCompression.swift

```diff
@@ -0,0 +1,100 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import Foundation
+import Compression
+import zlib
+
+/// Compresses the data into `ZLIB` data format.
+///
+/// The `Compression` library implements the zlib encoder at level 5 only. This compression level
+/// provides a good balance between compression speed and compression ratio.
+///
+/// The encoded format is the ZLIB Compressed Data Format as described in IETF RFC 1950
+/// https://datatracker.ietf.org/doc/html/rfc1950
+///
+/// - Parameter data: Source data to deflate
+/// - Returns:  The compressed data format.
+internal func zip(_ data: Data) -> Data? {
+    // 2 bytes header - defines the compression mode
+    // +---+---+
+    // |CMF|FLG|
+    // +---+---+
+    // ref. https://datatracker.ietf.org/doc/html/rfc1950#section-2.2
+    let header = Data([0x78, 0x5e])
```

Comment:
are these flags randomly generated ?

Reply:
This is coming from [mw99/DataCompression](https://github.com/mw99/DataCompression/blob/master/Sources/DataCompression/DataCompression.swift#L98) which is using the same algorithm defined in the Apple's `Compression` framework: [COMPRESSION_ZLIB](https://developer.apple.com/documentation/compression/compression_zlib)